vms/event/tests/test_model.py

```diff
@@ -0,0 +1,99 @@
+# third party
+
+# Django
+from django.db.models import Q
+from django.test.testcases import TestCase
+
+# local Django
+from event.models import Event
+from shift.utils import create_event_with_details
+
+
+class EventModelTests(TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    @staticmethod
+    def create_event():
+        event = ['event-name', '2050-05-24', '2050-05-28']
+        created_event = create_event_with_details(event)
+        return created_event
+
+    def test_valid_model_create(self):
+        event = ['event-name', '2050-05-24', '2050-05-28']
+        create_event_with_details(event)
+
+        # Check database for instance creation
+        self.assertNotEqual(len(Event.objects.all()), 0)
+
+        event_in_db = Event.objects.get(Q(name='event-name'))
+        # Verify correctness
+        self.assertEqual(event_in_db.name, event[0])
+        self.assertEqual(str(event_in_db.start_date), event[1])
+        self.assertEqual(str(event_in_db.end_date), event[2])
+
+    # Checks are missing from model class
```

Comment:
Raise issue  after confirmation in tomorrow's meeting.

Reply:
Workaround found:
model validation checks aren't performed until a `full_clean` is called explicitly from code/command-line, Forms perform `full_clean` on submit thats why the error messages are shown. Thus `model.save()` will save the instance unless there not-null constraint is broken.

Reply:
Does it make sense to use the workaround for now and file an issue to fix the bug later?

Reply:
Well its not a workaround per se, as I explained in the later part that model.save() doesnt actually executes the validation checks we need clean or full clean for it which is done by default in forms so there we can see an error while saving i.e.form.save but not in model.save. Ideally we should have a custom clean function in model to raise error while model save.